### ISSUE TITLE
allow_blessed default behaviour (throw exception) does not work as documented

### ISSUE THREAD

**Issue by wolfsage** [rurban/Cpanel-JSON-XS#105]
If ->convert_blessed is true but ->allow_blessed is not, an object with no string overloading or TO_JSON method should throw an exception.

This works in JSON::XS but not Cpanel::JSON::XS:

`$ cat wtf.pl`

````
#!/usr/bin/env perl

use strict;
use warnings;

use JSON::XS ();
use Cpanel::JSON::XS ();

my $json_xs = JSON::XS->new->convert_blessed;
my $cpanel_json_xs = Cpanel::JSON::XS->new->convert_blessed;

my $thing = bless {}, "CAT";

eval {
  print "json_xs: " . $json_xs->encode({ cat => $thing });
}; warn "json_xs: $@\n" if $@;

eval {
  print "cpanel_json_xs: " . $cpanel_json_xs->encode({ cat => $thing });
}; warn "cpanel_json_xs: $@\n" if $@;
````

`$ perl wtf.pl`
````
json_xs: encountered object 'CAT=HASH(0xe78da8)', but neither allow_blessed, convert_blessed nor allow_tags settings are enabled (or TO_JSON/FREEZE method missing) at wtf.pl line 15.

cpanel_json_xs: {"cat":null}
````

**Comment by rurban** [comment]
Fixed this bug, but there's still a stringification error.
Note that JOSN::XS cannot stringify either, which I fixed then.

**Comment by wolfsage** [comment]
Stringification overload doesn't appear to be supported by JSON::XS looking at the docs, actually

**Comment by wolfsage** [comment]
Also `convert_blessed($o3)` isn't what you mean is it? Shouldn't that be ->convert_blessed->encode($o3)`

**Comment by rurban** [comment]
Sure. Right, that's unsupported/broken. 
I.e. code for stringification is in JSON::XS, but broken because he mixed up refs.

**Comment by reyjrar** [comment]
Going to chime that I appreciate the work done on this PR.  Is there anything I can do to help?

We're still in a position where some code is using JSON or JSON::XS, but newer code is converted to JSON::MaybeXS which favors Cpanel::JSON::XS.  This difference in blessed object serialization just creeped up in our test suite.  The tests passed with JSON::XS loaded, but they failed since Cpanel::JSON::XS returned `null` for an object when `->allowed_blessed()` wasn't set, but `->convert_blessed()` was enabled.

**Comment by rurban** [comment]
So I digged through the code again and compared to JSON::XS. Some part of the documentation is also wrong:

I added this senetence to the allow_blessed section:
```
If C<$enable> is false (the default), then C<encode> will throw an
exception when it encounters a blessed object without a C<TO_JSON> method.
```

i.e. even JSON::XS does not check for allow_blessed on an objectref with a TO_JSON method.

**Comment by rurban** [comment]
Fixed with 1ea8fdffe52c5319a24e26e4f068f92350243aa2
and closed with fced48086be476551b489a16fb268fef0b493772 (Release 4.20)
